### ISSUE TITLE
Support European Portuguese 🇵🇹

### ISSUE THREAD

**Issue by duarteocarmo** [EuroEval/EuroEval#1040]
I'd like to take on the challenge of adding Portuguese to EuroEval. The most challenging aspect of this is going to be that European Portuguese is slightly different than the Brazilian version. For example, [belebele](https://huggingface.co/datasets/facebook/belebele/viewer/por_Latn) has mostly Brazilian Portuguese examples. 

Potential European Portuguese datasets to fit into the existing tasks:

### 🇵🇹 Portuguese (PT-PT) Datasets

| Task                     | Dataset                                                                 | Notes                                |
|--------------------------|-------------------------------------------------------------------------|--------------------------------------|
| Sentiment Classification | [PORTULAN/extraglue sst2](https://huggingface.co/datasets/PORTULAN/extraglue/viewer/sst2_pt-PT) | ✅ Covered by [#1043](https://github.com/EuroEval/EuroEval/issues/1043) |
| Named Entity Recognition | https://huggingface.co/datasets/Linguateca/harem                                                                     | ✅ Covered by [#1043](https://github.com/EuroEval/EuroEval/issues/1043)           |
| Linguistic Acceptability | https://github.com/UniversalDependencies/UD_Portuguese-Bosque                                                                    |   ✅ Covered by [#1043](https://github.com/EuroEval/EuroEval/issues/1043)            |
| Reading Comprehension    | [PORTULAN/extraglue boolq](https://huggingface.co/datasets/PORTULAN/extraglue/viewer/boolq_pt-PT) |  ✅ Covered by [#1043](https://github.com/EuroEval/EuroEval/issues/1043) |
| Summarization            | https://huggingface.co/datasets/duarteocarmo/sum-pt-publico                                                                       | ✅ Covered by [#1043](https://github.com/EuroEval/EuroEval/issues/1043)             |
| Knowledge                | [opengpt-x_mmlux](https://huggingface.co/datasets/LumiOpen/opengpt-x_mmlux) | ✅ Covered by [#1043](https://github.com/EuroEval/EuroEval/issues/1043) |
| Common-sense Reasoning   | [opengpt-x_goldenswagx](https://huggingface.co/datasets/LumiOpen/opengpt-x_goldenswagx/viewer/PT-PT) | ✅ Covered by [#1043](https://github.com/EuroEval/EuroEval/issues/1043) |

I've created an initial PR, happy to get any feedback!!

**Comment by saattrupdan** [comment]
Great stuff, thanks @duarteocarmo!

As for the datasets, are these in European Portuguese?

MMLU-pt: https://huggingface.co/datasets/LumiOpen/opengpt-x_mmlux

Goldenswag-pt: https://huggingface.co/datasets/LumiOpen/opengpt-x_goldenswagx/viewer/PT-PT

We can extend ScaLA to Portuguese as a linguistic acceptability benchmark, using the Portuguese universal dependencies. Can you check if this is European Portuguese?

https://github.com/UniversalDependencies/UD_Portuguese-Porttinari/blob/master/pt_porttinari-ud-dev.conllu

As for NER, again, can you check whether this is European Portuguese?

https://huggingface.co/datasets/tner/multinerd/viewer/pt

**Comment by saattrupdan** [comment]
As for the reading comprehension dataset, it needs to be an extractive question answering dataset, and not a multiple choice one. 

I'm currently making a massively multilingual reading comprehension dataset - could you have a look at the Portuguese survey with the link below and see if it's European Portuguese?

https://www.saattrupdan.com/multiwikiqa-feedback

**Comment by saattrupdan** [comment]
Lastly, as for summarisation, can you see if this one is in European Portuguese?

https://huggingface.co/datasets/csebuetnlp/xlsum/viewer/portuguese

**Comment by duarteocarmo** [comment]
> As for the reading comprehension dataset, it needs to be an extractive question answering dataset, and not a multiple choice one.
> 
> I'm currently making a massively multilingual reading comprehension dataset - could you have a look at the Portuguese survey with the link below and see if it's European Portuguese?
> 
> https://www.saattrupdan.com/multiwikiqa-feedback

@saattrupdan, ah ok - I've made in my PR something pretty similar to [Belebele](https://euroeval.com/datasets/spanish/#unofficial-belebele-es) for now, but can remove if you think bad idea.

**Comment by duarteocarmo** [comment]
> Great stuff, thanks [@duarteocarmo](https://github.com/duarteocarmo)!
> 
> As for the datasets, are these in European Portuguese?
> 
> MMLU-pt: https://huggingface.co/datasets/LumiOpen/opengpt-x_mmlux

Nice catch. Added to description. 

> Goldenswag-pt: https://huggingface.co/datasets/LumiOpen/opengpt-x_goldenswagx/viewer/PT-PT

Another nice catch. Added to description. 

> We can extend ScaLA to Portuguese as a linguistic acceptability benchmark, using the Portuguese universal dependencies. Can you check if this is European Portuguese?
> 
> https://github.com/UniversalDependencies/UD_Portuguese-Porttinari/blob/master/pt_porttinari-ud-dev.conllu

This is brazilian portuguese.


> 
> As for NER, again, can you check whether this is European Portuguese?
> 
> https://huggingface.co/datasets/tner/multinerd/viewer/pt

This is also brazilian portuguese

**Comment by duarteocarmo** [comment]
> As for the reading comprehension dataset, it needs to be an extractive question answering dataset, and not a multiple choice one.
> 
> I'm currently making a massively multilingual reading comprehension dataset - could you have a look at the Portuguese survey with the link below and see if it's European Portuguese?
> 
> https://www.saattrupdan.com/multiwikiqa-feedback

I am no linguist. But this seems like a bit of a mix - but a lot of Brazilian PT in there.

**Comment by duarteocarmo** [comment]
> Lastly, as for summarisation, can you see if this one is in European Portuguese?
> 
> https://huggingface.co/datasets/csebuetnlp/xlsum/viewer/portuguese

That's also a mix. But mostly brazil pt. Here's a very bad heuristic: 

| Concept | pt-BR token | pt-PT token |
|---------|-------------|-------------|
| bus     | **ônibus**  | **autocarro** |
| juice   | **suco**    | **sumo** |
| mobile phone | **celular** | **telemóvel** |
| train   | **trem**    | **comboio** |

Usage is country-specific in news, emails, social media, etc.  
**Rule of thumb:**  
*if a text contains any left-hand word ⇒ tag `pt-BR`; any right-hand word ⇒ tag `pt-PT`.*

**Comment by duarteocarmo** [comment]
If you agree, I'll add MMLU and GoldenSwag to the PR

**Comment by duarteocarmo** [comment]
Let me know @saattrupdan

**Comment by duarteocarmo** [comment]
@saattrupdan  - updated the PR with two new datasets!

**Comment by saattrupdan** [comment]
As for ScaLA, this seems to be European Portuguese - the samples with source=CETEMPúblico:

https://github.com/UniversalDependencies/UD_Portuguese-Bosque

**Comment by duarteocarmo** [comment]
> As for ScaLA, this seems to be European Portuguese - the samples with source=CETEMPúblico:
> 
> https://github.com/UniversalDependencies/UD_Portuguese-Bosque

@saattrupdan  nice catch - will add.

**Comment by saattrupdan** [comment]
And could you check these datasets for a European Portuguese NER dataset?

https://huggingface.co/datasets?task_categories=task_categories:token-classification&language=language:pt&sort=trending

Same thing for this list regarding summarisation:

https://huggingface.co/datasets?task_categories=task_categories:summarization&language=language:pt&sort=trending

**Comment by saattrupdan** [comment]
Found this classifier between Brazilian and European Portuguese, which we could use to separate the reading comprehension dataset associated with the survey that I linked to above:

https://huggingface.co/liaad/PtVId

**Comment by duarteocarmo** [comment]
> Found this classifier between Brazilian and European Portuguese, which we could use to separate the reading comprehension dataset associated with the survey that I linked to above:
> 
> https://huggingface.co/liaad/PtVId

This is a super nice find!!!

**Comment by duarteocarmo** [comment]
@saattrupdan  - I think I found the perfect one for NER. What do you think? 

https://huggingface.co/datasets/Linguateca/harem

**Comment by saattrupdan** [comment]
> @saattrupdan  - I think I found the perfect one for NER. What do you think? 
> 
> https://huggingface.co/datasets/Linguateca/harem

This looks good! The paper they mention is called "BERTimbau: Pretrained BERT Models for Brazilian Portuguese" though, but it seems like in their paper they mention that half of the dataset is from European Portuguese sources. Are you able to check if there are European Portuguese samples in there?

**Comment by duarteocarmo** [comment]
> > [@saattrupdan](https://github.com/saattrupdan)  - I think I found the perfect one for NER. What do you think?
> > https://huggingface.co/datasets/Linguateca/harem
> 
> This looks good! The paper they mention is called "BERTimbau: Pretrained BERT Models for Brazilian Portuguese" though, but it seems like in their paper they mention that half of the dataset is from European Portuguese sources. Are you able to check if there are European Portuguese samples in there?

Very weird. Linguateca is a portuguese organization. Let me recheck and I'll get back to you. We are getting a very nice complete set here! Excited :)

**Comment by duarteocarmo** [comment]
(I'm updating the table in the issue description as we go)

**Comment by duarteocarmo** [comment]
> > > [@saattrupdan](https://github.com/saattrupdan)  - I think I found the perfect one for NER. What do you think?
> > > https://huggingface.co/datasets/Linguateca/harem
> > 
> > 
> > This looks good! The paper they mention is called "BERTimbau: Pretrained BERT Models for Brazilian Portuguese" though, but it seems like in their paper they mention that half of the dataset is from European Portuguese sources. Are you able to check if there are European Portuguese samples in there?
> 
> Very weird. Linguateca is a portuguese organization. Let me recheck and I'll get back to you. We are getting a very nice complete set here! Excited :)

@saattrupdan  ok, so HAREM actually has a mix of portuguese from Portugal, Brazil, and from other places like Macau and Mozambique - I think we can use it. We can filter by ORIGEM = PORTUGAL, or run the classifier. 

https://huggingface.co/datasets/liaad/Harem/raw/main/raw/first/ColeccaoDouradaHAREM.txt

**Comment by duarteocarmo** [comment]
Also, I tested the Vid classifier and it's pretty good. It's a bit strong on the confidences (like any other bert fine tune) but should serve us well.

**Comment by saattrupdan** [comment]
> I think we can use it. We can filter by ORIGEM = PORTUGAL, or run the classifier.

Great! I don't think we need the classifier if we have gold labels in the data 🙂

Also, since you said the summarisation dataset is a mix of Brazilian and European Portuguese, we could probably use the classifier there. We could include that as part of the dataset generation script.

With that, as well as the upcoming reading comprehension dataset, we should be good! The multiple choice reading comprehension dataset can also be included, but the "regular" ones are preferred, as we can also evaluate encoder models on those. 

All the datasets should then be official, except the multiple choice reading comprehension dataset.

**Comment by duarteocarmo** [comment]
Ok, added Scala. Now onto summarization and NER.

**Comment by duarteocarmo** [comment]
For the summarization, I've ran the classifier over `xlsum` in the portuguese split, but it's from the BBC website brazil. 

In a sample of 6000, I got 

PT-BR    5631
PT-PT     369

The PT-PT are likely false positives IMO.

**Comment by duarteocarmo** [comment]
Checking the NER and what I can do there for now.

**Comment by duarteocarmo** [comment]
Ok, I have a plan for the NER stuff. For the summarization @saattrupdan  - wdyt of this? 
https://huggingface.co/datasets/dennlinger/eur-lex-sum

**Comment by saattrupdan** [comment]
> Ok, I have a plan for the NER stuff. For the summarization [@saattrupdan](https://github.com/saattrupdan) - wdyt of this? https://huggingface.co/datasets/dennlinger/eur-lex-sum

Hm, I'm a bit sceptical, as it is very legal heavy.

There is [another one](https://huggingface.co/datasets/GEM/wiki_lingua), based on [WikiHow](https://pt.wikihow.com/). Are Portuguese WikiHow articles in European Portuguese?

**Comment by saattrupdan** [comment]
Also found [this one](https://www.linguateca.pt/Repositorio/PCSC/), published in [this paper](http://www.lrec-conf.org/proceedings/lrec2014/pdf/187_Paper.pdf). The title looks promising: "Priberam Compressive Summarization Corpus: A New Multi-Document Summarization Corpus for European Portuguese".

It would need to be set up as a Hugging Face dataset, but that could be done in the dataset generation script.

**Comment by duarteocarmo** [comment]
Yes - I came across that one, but it's only 800 docs or so, should we still add?

**Comment by duarteocarmo** [comment]
@saattrupdan  - added NER as well from Harem to the PR. Just the summarization missing - should we go with [this](https://priberam.com/labs/resources/pcsc/)?

**Comment by saattrupdan** [comment]
I think we should just go with the one with 800 documents yeah. Perhaps a 64/32/704 train/val/test split would do.

**Comment by duarteocarmo** [comment]
@saattrupdan looks like it's not really 800. I looked into the dataset. 

Let me explain and lmk what you think. 

### Option 1 — Multi-document → Summary (original setup)
- Each topic has 10 news articles and 2 human-written summaries.
- We concatenate all 10 docs as input and pair with each summary.
- → 160 examples total (80 topics × 2 summaries).
- ✅ Clean supervision — summaries were written with access to all documents.
- ❌ Very limited data, likely not enough for training.. Wdyt?

### Option 2 — Single-document → Summary (approximate setup)
- Each of the 800 documents is paired with the topic-level summary.
- → 800 examples (1 summary per topic) or 1600 (if using both).
- ✅ More data for model training.
- ❌ BUT THIS IS NOISY supervision — summaries may include info not present in the individual doc...

Wdyt? Should we still go with option 2?

**Comment by saattrupdan** [comment]
@duarteocarmo I think concatenating the 10 documents would result in way too long inputs.

I had a look at a few of the news articles, and it seems that they follow the same pattern as most news articles: the first paragraph after the heading actually summarises the article itself. This is a common "trick" to creating summarisation datasets, by just using that as the summary, and including everything after that (as well as the title) as the article.

If we do this for all the articles, then we should have 800 pretty good summaries. It's a shame to throw away the human written ones, but I just don't think they're that useful for us when they're multi-document summaries, unfortunately.

**Comment by duarteocarmo** [comment]
Nice - if you're ok with that trick then I have a slightly better idea - will post here soon :)

**Comment by duarteocarmo** [comment]
@saattrupdan 

Ok, I filtered [this dataset](https://huggingface.co/datasets/eduagarcia/cc_news_pt_v2)  by the `publico.pt` url. Which is the politiken from portugal (kinda). Maybe we can apply trick to it? It's certainly higher quality in terms of the articles - and I am pretty sure they are all portuguese from portugal! 

Let me know if you're ok w/ this.

**Comment by saattrupdan** [comment]
I'm not sure if it's just the datasets viewer, but are the paragraphs correctly separated (e.g., by newlines)? Otherwise the trick will be a bit difficult to apply 😛

Also, in case they *are* split correctly, can you manually check a couple of them, to see if they make sense as summaries?

**Comment by duarteocarmo** [comment]
I am not sure the priberam dataset always follows that principle of having a first paragraph separated also tbh.. 

Here are some examples of that one: https://gist.github.com/duarteocarmo/a07bf9171a4d56bdc9113d5a3a8a05be

My idea would be to take these ones https://huggingface.co/datasets/duarteocarmo/cc_news_publico/viewer/default/train?views%5B%5D=train&row=3 and the target text would be the first 2 sentences. 

Manually checking the first 2/3 rows this is not perfect, but pretty good. Those priberam ones are also good candidates, but they are not so clean in terms of formatting. I would go the publico one - but happy to go with priberam if you think better!

**Comment by duarteocarmo** [comment]
Here's the publico one in the euroeval format: https://huggingface.co/datasets/duarteocarmo/sum-pt-publico

**Comment by saattrupdan** [comment]
If you think the publico ones are pretty good, then I trust you - let's add that one 🙂

**Comment by duarteocarmo** [comment]
@saattrupdan ok perfect, I think the PR is ready for a review!

**Comment by saattrupdan** [comment]
I've filtered the reading comprehension dataset that I mentioned above, using the classifier. Can you check if the samples are indeed European Portuguese, in the following survey?

[saattrupdan.com/multiwikiqa-feedback](https://www.saattrupdan.com/multiwikiqa-feedback)

If so, I can add that as the official Portuguese reading comprehension dataset after this PR 🙂

**Comment by duarteocarmo** [comment]
> [saattrupdan.com/multiwikiqa-feedback](https://www.saattrupdan.com/multiwikiqa-feedback)

Ok, I think we are ok, largely it's portuguese from portugal. 

[Notes here](https://docs.google.com/spreadsheets/d/16FGtIpsMbVgXwR9zwECOeAmmkXbnghWmlNw_4u9lZtg/edit?usp=sharing)

**Comment by duarteocarmo** [comment]
ChatGPT is really bad at this btw 😂
